src/agent/src/command_handler_utils.cpp

```diff
@@ -13,48 +13,51 @@ namespace
     template<typename ExecuteFunction>
     boost::asio::awaitable<void> ExecuteCommandTask(ExecuteFunction executeFunction,
                                                     module_command::CommandEntry commandEntry,
-                                                    std::shared_ptr<module_command::CommandExecutionResult> result,
-                                                    std::shared_ptr<bool> commandCompleted,
-                                                    std::shared_ptr<boost::asio::steady_timer> timer)
+                                                    std::shared_ptr<module_command::CommandExecutionResult> result)
     {
         try
         {
             *result = co_await executeFunction(commandEntry.Command, commandEntry.Parameters);
-            *commandCompleted = true;
-            timer->cancel();
         }
         catch (const std::exception& e)
         {
-            result->ErrorCode = module_command::Status::FAILURE;
-            result->Message = "Error during command execution: " + std::string(e.what());
+            if (result->ErrorCode == module_command::Status::UNKNOWN)
+            {
+                result->ErrorCode = module_command::Status::FAILURE;
+                result->Message = "Error during command execution: " + std::string(e.what());
+            }
         }
     }
 
-    boost::asio::awaitable<void> TimerTask(std::shared_ptr<boost::asio::steady_timer> timer,
-                                           std::shared_ptr<module_command::CommandExecutionResult> result,
-                                           std::shared_ptr<bool> commandCompleted)
+    boost::asio::awaitable<void> TimerTask(std::shared_ptr<module_command::CommandExecutionResult> result)
     {
         try
         {
+            const auto timeout = std::chrono::minutes(60);
```

Comment:
Maybe we can use a `constexpr` for this (60).

Reply:
Done!